### PR TITLE
* #11: Added swagger.

### LINKED ISSUE #11: Add swagger for API documenation
## Problem

We want to have API documentation generated with swagger.

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #11

### CODE REVIEW

**Comment by AnastasiiaSergienko** [comment]
This is a simple implementation where the swagger files were generated manually with `swag init` command. It can be possible to automate it in future, but maybe we start with something simple

**Comment by jakobbraun** [comment]
> This is a simple implementation where the swagger files were generated manually with `swag init` command. It can be possible to automate it in future, but maybe we start with something simple

Adding it to the build should not be very much effort. There is a maven plugin for that: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen-maven-plugin/README.md

If you don't want to do it in this PR, please create a ticket.

File: main.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package main
+
+import exasol_rest_api "main/pkg/exasol-rest-api"
+
+// @title Exasol REST API
+// @version 0.1.0
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply main.go:6]: Different solution:
set version to something like `>>>project.version<<<`
replace this in the generated files by the project version from go file
(inspired from here: https://stackoverflow.com/questions/57821746/use-artifact-version-number-in-openapi-spec-definition)

**Review by jakobbraun — CHANGES_REQUESTED** [review]

**Comment by jakobbraun** [comment]
add developer guide
add script for building doc

File: README.md [hunk 2]
```diff
@@ -2,3 +2,17 @@
 
 Exasol REST API is an extension for the Exasol database that provides the ability to interact with the database via REST API endpoints.
 **This project is at an early development stage.**
+
+
+## Information for Users
+
+## Information for Developers
```
Comment (jakobbraun) [inline README.md:9]:
```suggestion
```

**Review by jakobbraun — CHANGES_REQUESTED** [review]

**Review by jakobbraun — APPROVED** [review]

**Pull request merged**